testutils/file_utils.go

```diff
@@ -37,7 +40,77 @@ func BuildTempModule(tb testing.TB, dir string) string {
 	if tb.Failed() {
 		tb.Fatalf("failed to build temporary module for testing")
 	}
-	return modPath
+	return exePath
+}
+
+// BuildTempModuleWithFirstRun will attempt to build the module in the provided directory and put the
+// resulting executable binary into a temporary directory. After building, it will also copy "meta.json"
+// and "first_run.sh" into the same temporary directory. It is assumed that these files are in the
+// provided module directory. If successful, this function will return the path to the executable binary.
+func BuildTempModuleWithFirstRun(tb testing.TB, modDir string) string {
+	// tb.Helper()
+
+	exePath := BuildTempModule(tb, modDir)
+	exeDir := filepath.Dir(exePath)
+
+	type copyOp struct {
+		src string
+		dst string
+	}
+
+	for _, cp := range []copyOp{
+		// TODO(RSDK-9151): Having a `meta.json` in the testmodule directory results in
+		// unintended behavior changes across our test suite. To avoid this, we name the
+		// file `first_run_meta.json` in the source test module directory and rename it in the
+		// destination directory that contains the built executable. This is fine for now
+		// but we should investigate and understand why this happens.
+		{"first_run_meta.json", "meta.json"},
+		{"first_run.sh", "first_run.sh"},
+	} {
+		resolvedModDir := utils.ResolveFile(modDir)
+
+		//nolint:gosec
+		dir, err := os.Open(resolvedModDir)
+		if err != nil {
+			tb.Fatal(err)
+		}
+		files, err := dir.ReadDir(-1)
+		if err != nil {
+			tb.Fatal(err)
+		}
+
+		tb.Log("files in mod directory:")
+		for _, file := range files {
+			tb.Log(file)
+		}
+
```

Comment:
this function can probably be cleaned up

Reply:
oh yeah, we don't need it anymore